### PR TITLE
Add new text size scale, deprecate old sizes and Heading

### PR DESCRIPTION
**Pull request by markdalgleish**
Fixes RNBW-4399
Figma link: https://www.figma.com/file/wFXhL8ZcaBnIlWSSQTLWPA/RDS-Core?node-id=12%3A222

## What changed (plus any additional context for devs)

New standard text sizes are now available on the design system `Text` component to align with new standards from Figma. Note that line height variants form part of the `size` scale.

- `11pt`
- `12pt`
- `13pt`, `13pt / 135%`, `13pt / 150%`
- `15pt`, `15pt / 135%`, `15pt / 150%`
- `17pt`, `17pt / 135%`, `17pt / 150%`
- `20pt`, `20pt / 135%`, `20pt / 150%`
- `22pt`
- `26pt`
- `30pt`
- `34pt`
- `44pt`

All new heading sizes are now provided via `Text`. The `Heading` component is now marked as deprecated and will be removed in a future PR.

Default `size` values for `Text` (`16px`) and `Heading` (`20px`) have been removed since the previous defaults have been deprecated.

To make future migration work easier, the default `weight` value for `Heading` has been removed, forcing all usages to inline the default value of `"heavy"`.

### Heading/Text size migration guide

All existing sizes have been renamed to make it very clear that they shouldn't be used going forwards while maintaining support for existing usages. To align with the new naming convention, the old sizes also include their line height values.

**Heading migration guide**

Note that `weight` no longer has a default value. If no weight was defined, it should be "heavy".

- `<Heading>` -> `<Heading size="20px / 22px (Deprecated)" weight="heavy">`
- `<Heading size="18px">` -> `<Heading size="18px / 21px (Deprecated)" weight="heavy">`
- `<Heading size="20px">` -> `<Heading size="20px / 22px (Deprecated)" weight="heavy">`
- `<Heading size="23px">` -> `<Heading size="23px / 27px (Deprecated)" weight="heavy">`
- `<Heading size="26px">` -> `<Heading size="26px / 30px (Deprecated)" weight="heavy">`
- `<Heading size="28px">` -> `<Heading size="28px / 33px (Deprecated)" weight="heavy">`
- `<Heading size="30px">` -> `<Heading size="30px / 34px (Deprecated)" weight="heavy">`
- `<Heading size="34px">` -> `<Heading size="34px / 41px (Deprecated)" weight="heavy">`
- `<Heading size="44px">` -> `<Heading size="44px / 53px (Deprecated)" weight="heavy">`

**Text migration guide**
- `<Text>` -> `<Text size="16px / 22px (Deprecated)">`
- `<Text size="11px">` -> `<Text size="11px / 13px (Deprecated)">`
- `<Text size="12px">` -> `<Text size="12px / 14px (Deprecated)">`
- `<Text size="14px">` -> `<Text size="14px / 19px (Deprecated)">`
- `<Text size="15px">` -> `<Text size="15px / 21px (Deprecated)">`
- `<Text size="16px">` -> `<Text size="16px / 22px (Deprecated)">`
- `<Text size="18px">` -> `<Text size="18px / 27px (Deprecated)">`
- `<Text size="20px">` -> `<Text size="20px / 24px (Deprecated)">`
- `<Text size="23px">` -> `<Text size="23px / 27px (Deprecated)">`

## What to test

Check that text is still rendering correctly in the app. TypeScript is doing most of the heavy lifting here, but any changes to JS files require a bit of extra scrutiny. The list of deprecated text sizes above is helpful when checking the diff, but I've also added some additional code to throw an error in dev mode when the requested text size is missing so it should get picked up in e2e tests.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/components/ens-profile/RecordTags/RecordTags.tsx [hunk 1]
```diff
@@ -81,17 +82,24 @@ export default function RecordTags({
   );
 }
 
+const tagSizes = selectTextSizes(
```
Comment (jxom) [inline src/components/ens-profile/RecordTags/RecordTags.tsx:85]: nice

File: src/design-system/typography/typeHierarchy.ts [hunk 2]
```diff
@@ -1,6 +1,36 @@
+/**
+ * ⚠️ IMPORTANT NOTE ⚠️
```
Comment (jxom) [inline src/design-system/typography/typeHierarchy.ts:2]: Maybe also add a link to the Figma spec here so devs can visually reference what is in this file?

File: src/design-system/components/Text/useTextStyle.ts [hunk 3]
```diff
@@ -26,6 +26,16 @@ export function useTextStyle({
           .join(', ')}`
       );
     }
+
+    if (!textSizes[size]) {
```
Comment (estrattonbailey) [inline src/design-system/components/Text/useTextStyle.ts:30]: Love this. Do we need an `if (__DEV__) {` check like you did above?

**Review by estrattonbailey — APPROVED** [review]
Beautiful 💪

**Review by jxom — APPROVED** [review]

File: src/design-system/components/Text/useTextStyle.ts (hunk 3 above)
Comment (markdalgleish) [reply src/design-system/components/Text/useTextStyle.ts:30]: This is actually nested inside one large `if (__DEV__) {` block, it's just not visible in this diff—but I can see that makes it less clear so I might update it.

File: src/design-system/typography/typeHierarchy.ts (hunk 2 above)
Comment (markdalgleish) [reply src/design-system/typography/typeHierarchy.ts:2]: I'm not sure how set in stone the Figma URL is, so I might come back to this one.

File: src/design-system/components/Text/useTextStyle.ts (hunk 3 above)
Comment (estrattonbailey) [reply src/design-system/components/Text/useTextStyle.ts:32]: 🤦 my b!

**Pull request merged**